src/NHibernate.Test/NHSpecificTest/GH1712/Fixture.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using System.Linq;
+using NUnit.Framework;
+
+namespace NHibernate.Test.NHSpecificTest.GH1712
+{
+	[TestFixture]
+	public class Fixture : BugTestCase
+	{
+		protected override void OnSetUp()
+		{
+			using (var session = OpenSession())
+			using (var tx = session.BeginTransaction())
+			{
+				session.Save("GenericEntityWithGuid", new GenericEntity<Guid>(),
```

Comment:
@fredericDelaporte let me know if you want me to add the other key types (bytes, numeric types, time/date types, etc.)